backend/app/src/main/kotlin/io/tolgee/api/v2/controllers/NamespaceController.kt

```diff
@@ -0,0 +1,49 @@
+package io.tolgee.api.v2.controllers
+
+import io.swagger.v3.oas.annotations.Operation
+import io.swagger.v3.oas.annotations.tags.Tag
+import io.tolgee.api.v2.hateoas.key.namespace.NamespaceModel
+import io.tolgee.api.v2.hateoas.key.namespace.NamespaceModelAssembler
+import io.tolgee.controllers.IController
+import io.tolgee.model.key.Namespace
+import io.tolgee.security.apiKeyAuth.AccessWithApiKey
+import io.tolgee.security.project_auth.AccessWithAnyProjectPermission
+import io.tolgee.security.project_auth.ProjectHolder
+import io.tolgee.service.key.NamespaceService
+import org.springdoc.api.annotations.ParameterObject
+import org.springframework.data.domain.Pageable
+import org.springframework.data.web.PagedResourcesAssembler
+import org.springframework.data.web.SortDefault
+import org.springframework.hateoas.PagedModel
+import org.springframework.web.bind.annotation.CrossOrigin
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RestController
+
+@Suppress("MVCPathVariableInspection")
+@RestController
+@CrossOrigin(origins = ["*"])
+@RequestMapping(
+  value = [
+    "/v2/projects/{projectId}/namespaces",
```

Comment:
We are stoping to use these endpoints and probably will deprecate then, since now you can parse project id from the PAK. But I should probably add it for now... 🤔

Reply:
If the long term goal is to deprecate the old way of accessing projects, not including parameter-less endpoints for new endpoints is a reasonable thing to do.

That being said, I have mixed feelings about having to parse the API key to know the project to query; similar to how most OAuth-enabled APIs don't require pulling the user id from the token somehow and have a special `me` identifier to access the current user.

Reply:
Currently the whole 2 endpoints thing make whole thing too complicated. So we were thinking about this approach to provide some compromise between requiring users to specify the project ID and having 2 endpoints for each project operation...